test/common.py

```diff
@@ -1336,17 +1336,17 @@ def in_dir(self, *pathelems):
   def add_pre_run(self, code):
     assert not self.get_setting('MINIMAL_RUNTIME')
     create_file('prerun.js', 'Module.preRun = function() { %s }\n' % code)
-    self.emcc_args += ['--pre-js', 'prerun.js']
+    self.emcc_args += ['--pre-js', 'prerun.js', '-sINCOMING_MODULE_JS_API=[preRun]']
```

Comment:
The problem with doing this here is that then the test doesn't have any way to add more to `INCOMING_MODULE_JS_API`.

We currently don't have any way to make settings like `INCOMING_MODULE_JS_API` additive.

I've tried to make them additive by default (which seems like the right behaviour in almost all cases I can think of): https://github.com/emscripten-core/emscripten/pull/19938.   But that PR got hung up with disagreement about how / if to change the default behaviour.

Reply:
That is, either that specific test can be written to duplicate the `-sINCOMING_MODULE_JS_API=[preRun,theothersymbol]`, or `-sINCOMING_MODULE_JS_API=[preRun]` be hoisted from `add_pre_run()` to all the tests.